security/openpgp.ksy

```diff
@@ -0,0 +1,508 @@
+meta:
+  id: pgp
+  title: OpenPGP Message Format
+  license: MIT
+  file-extension: 
+    - gpg
+    - pub
+    - pgp
+  endian: be
+doc: The OpenPGP Message Format is a format to store encryption and signature keys for emails.
+doc-ref: https://tools.ietf.org/html/rfc4880
+seq:
+ - id: packets
+   type: packet
+   repeat: eos
+types:
+  packet:
+    -webide-representation: '{packet_type_old}'
+    seq:
+      - id: one
+        type: b1
+      - id: new_packet_format
+        type: b1
+      - id: packet_type_new
+        type: b6
+        if: new_packet_format
+        enum: packet_tags
+      - id: packet_type_old
+        type: b4
+        if: not new_packet_format
+        enum: packet_tags
+      - id: length_type
+        type: b2
+        if: not new_packet_format
+      - id: body
+        type: 
+          switch-on: new_packet_format
+          cases:
+            #true: new_packet
+            false: old_packet
+            
+  old_packet:
+    seq:
+      - id: length
+        type: 
+          switch-on: _parent.length_type
+          cases:
+            0: u1
```

Comment:
Consider creating a enum.

Reply:
What would be the advantage in this case?

Reply:
No real advantage, just a more fancier look.